protected/humhub/components/behaviors/AccessControl.php

```diff
@@ -220,4 +223,14 @@ protected function loginRequired()
 
         return false;
     }
+
+    /**
+     * @return Response Redirect user to force to change password
+     */
+    protected function mustChangePassword()
+    {
+        if (!Yii::$app->user->isMustChangePasswordUrl()) {
+            return Yii::$app->getResponse()->redirect(Yii::$app->user->mustChangePasswordUrl);
```

Comment:
Maybe check if its a pjax or full page load. In case of simple ajax call it should just return a forbidden code. Maybe this would also partly solve the PollController issue?

Reply:
Please see my investigation with code:

```
if (Yii::$app->getRequest()->getIsAjax()) {
    $this->forbidden();
}
```
<img width="1137" alt="investigate_ajax_restriction" src="https://user-images.githubusercontent.com/6995524/98516406-860a2400-227d-11eb-8734-f75cd69e128d.png">

---

I.e. with current code if user becomes blocked with option "must change password" then auto redirected to `/user/must-change-password` on ajax request, but if we will add the forbidden exception code then user will stay on the same page without redirect, do you think it is better? If yes, I will add the code.

> Maybe this would also partly solve the PollController issue?

I didn't see the poll requests are locked by access controller, they continue work with 200 code.